utilities/network.py

```diff
@@ -1038,23 +1041,189 @@ def get_cluster_cni_type(admin_client):
+        bool: True if status is SUCCEEDED and (if policy provided) configuration matches
+
+    Raises:
+        ResourceMismatch: If interface configuration doesn't match policy (will be retried by TimeoutSampler)
+    """
+    current_status = sriov_node_network_state.instance.status.syncStatus
+    if current_status != SriovNetworkNodePolicy.Status.SUCCEEDED:
+        return False
+
+    if policy:
+        _verify_sriov_interface_config(
+            sriov_node_network_state=sriov_node_network_state,
+            policy=policy,
+            admin_client=admin_client,
+        )
+
+    return True
+
+
+def wait_for_ready_sriov_nodes(
+    snns: list[SriovNetworkNodeState],
+    admin_client: DynamicClient,
+    policy: SriovNetworkNodePolicy = None,
+):
+    """
+    Wait for SR-IOV nodes to be ready and verify interface configuration matches policy.
+
+    This function is used in two scenarios:
+    1. After policy creation: Waits for SUCCEEDED status and verifies configuration (policy provided)
+    2. After policy teardown: Waits for SUCCEEDED status only (policy=None, policy already deleted)
+
+    Args:
+        snns: List of SriovNetworkNodeState objects
+        admin_client: Kubernetes admin client (required for Node access)
+        policy: SriovNetworkNodePolicy object to verify configuration against.
+                None during teardown when policy is already deleted.
+
+    Raises:
+        TimeoutExpiredError: If sync status doesn't reach SUCCEEDED state or configuration doesn't match
+        ResourceMismatch: If interface configuration doesn't match policy (on setup)
+    """
+    for sriov_node_network_state in snns:
+        status_msg = f"to be {SriovNetworkNodePolicy.Status.SUCCEEDED}"
+        if policy:
+            status_msg += f" and configured per policy {policy.name}"
+        LOGGER.info(f"Waiting for node {sriov_node_network_state.name} SriovNetworkNodeState {status_msg}")
+
+        sampler = TimeoutSampler(
+            wait_timeout=1000,
+            sleep=5,
+            func=_check_sriov_node_ready_and_configured,
+            sriov_node_network_state=sriov_node_network_state,
+            policy=policy,
+            admin_client=admin_client,
+            exceptions_dict={ResourceMismatch: []} if policy else {},
+        )
+        try:
+            for sample in sampler:
+                if sample:
+                    success_msg = (
+                        f"Node {sriov_node_network_state.name} SriovNetworkNodeState is"
+                        f" {SriovNetworkNodePolicy.Status.SUCCEEDED}"
                     )
-                    raise
+                    if policy:
+                        success_msg += " and configuration verified"
+                    LOGGER.info(success_msg)
+                    break
+        except TimeoutExpiredError:
+            current_status = sriov_node_network_state.instance.status.syncStatus
+            if current_status != SriovNetworkNodePolicy.Status.SUCCEEDED:
```

Comment:
nit: whenever possible, use a positive condition (i.e here `current_status == SriovNetworkNodePolicy.Status.SUCCEEDED`)
makes it easier to follow the flow